onnxruntime/python/tools/quantization/quantize.py

```diff
@@ -435,7 +439,7 @@ def quantize_static(
         qdq_ops = list(QDQRegistry.keys())
         op_types_to_quantize = list(set(q_linear_ops + qdq_ops))
 
-    model = load_model_with_shape_infer(Path(model_input))
+    model = model_input if isinstance(model_input, onnx.ModelProto) else load_model_with_shape_infer(Path(model_input))
```

Comment:
do shape inference when model_input is ModelProto?

Reply:
Good call. I'll investigate more on this.

Reply:
Updatedd